packages/odata-service-inquirer/src/prompts/edmx/entity-helper.ts

```diff
@@ -73,13 +129,19 @@ export function getEntityChoices(
         } else {
             entitySets = convertedMetadata.entitySets;
         }
-
         entitySets.forEach((entitySet, index) => {
+            const mainEntityParameterName = getNavigationPropertyForParameterisedEntity(
+                entitySet,
+                convertedMetadata?.entityTypes
+            );
             const choice: ListChoiceOptions<EntityAnswer> = {
                 name: entitySet.name,
                 value: {
                     entitySetName: entitySet.name,
-                    entitySetType: entitySet.entityTypeName // Fully qualified entity type name
+                    entitySetType: entitySet.entityTypeName, // Fully qualified entity type name
+                    ...(mainEntityParameterName && {
```

Comment:
Could I suggest we keep the same style as the other property assignments. Seems odd to use spread on a string. It will should be undefined anyway.
```suggestion
                   mainEntityParameterName
```